### DISCUSSION TITLE
Manually setting syntax highlighting independent of file extension.

### DISCUSSION THREAD

**Discussion by bufferhe4d** [zed-industries/zed#7150] [Feature Requests]
I am using the language SageMath which ends with .sage. However, the syntax highlighting is exactly the same as Python.

Is it possible to enable a setting such that all files with .sage extension is highlighted according to Python highlighting rules?

**Comment by adn1995** [comment]
I found a solution in the [docs](https://zed.dev/docs/configuring-languages#file-associations).

Edit the `file_types` setting in `settings.json`:
```
"file_types": {
  "Python": ["sage"]
}
```
This should tell Zed to treat `.sage` files as Python.

**Reply by nmlt** [reply]
I had to restart zed to make this work.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
